pkg/apis/install/v1alpha1/type.go

```diff
@@ -0,0 +1,394 @@
+	Name ModuleName `json:"name,omitempty"`
+
+	// Number of desired pods. This is a pointer to distinguish between explicit
+	// zero and not specified. Defaults to 1.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// +optional
+	Image string `json:"image,omitempty"`
+}
+
+type Images struct {
+	// karmada regisrty, default is "swr.ap-southeast-1.myhuaweicloud.com".
+	// all of karmada images is effect by the param.
+	// +optional
+	KarmadaRegistry string `json:"karmadaRegistry,omitempty"`
+
+	// karmada images version, default is "latest".
+	// all of karmada images is effect by the param.
+	// +optional
+	KarmadaVersion string `json:"karmadaVersion,omitempty"`
+
+	// the registry of kube-apiserver and kube-controller-manager, default is "k8s.gcr.io".
+	// +optional
+	KubeRegistry string `json:"kubeRegistry,omitempty"`
+
+	// the iamge version of kube-apiserver and kube-controller-manager.
+	// +optional
+	KubeVersion string `json:"kubeVersion,omitempty"`
+
+	// the registry of cfssl and kubectl, default is "docker.io".
+	// +optional
+	DockerIoRegistry string `json:"dockerIoRegistry,omitempty"`
+}
+
+type ETCD struct {
+	// TODO: support external etcd.
+	// StorageMode etcd data storage mode(emptyDir,hostPath, PVC), default "emptyDir"
+	// +optional
+	StorageMode string `json:"storageMode,omitempty"`
+
+	// If specified StorageMode is PVC, StorageClass must be set.
+	// +optional
+	StorageClass string `json:"storageClass,omitempty"`
+
+	// If specified StorageMode is PVC, Size must be set.
+	// +optional
+	Size string `json:"size,omitempty"`
+}
+
+type EndPointCfg struct {
+	// +optional
+	Kubeconfig string `json:"kubeconfig,omitempty"`
+
+	// The API endpoint of the member cluster. This can be a hostname,
+	// hostname:port, IP or IP:port.
+	// +optional
+	ControlPlaneEndpoint string `json:"controlPlaneEndpoint,omitempty"`
+
+	// SecretRef represents the secret contains mandatory credentials to access the member cluster.
+	// The secret should hold credentials as follows:
+	// - secret.data.token
+	// - secret.data.caBundle
+	// +optional
+	SecretRef *LocalSecretReference `json:"secretRef,omitempty"`
+}
+
+// the whole installtion process is be divide diffrent phase.
+//     Preflight: preparatory work, build cluster clientset pull karmada chart pkg.
+//     Installing: calculate chart values, install karmada chart by helm client.
+//     Waiting: wait to all of karmada pods to be ready.
+//     ControlPlaneReady: check karmada cluster is healthy.
+//     DeployFailedPhase: the karmada cluster is not healthy.
+// +enum
+type Phase string
+
+const (
+	PreflightPhase         Phase = "Preflight"
+	DeployingPhase         Phase = "Deploying"
+	WaitingPhase           Phase = "Waiting"
+	InstalledCRDPhase      Phase = "InstalledCRD"
+	ControlPlaneReadyPhase Phase = "Completed"
+)
```

Comment:
It should be defined in more detail, example install crd:
InstallingCRDPhase ->InstalledCRDPhase ->InstalledFailedCRDPhase

Reply:
I think we can remove the field，it make no difference.
we can discuss it the next meeting.